### DISCUSSION TITLE
Features of IPAM enhancements

### DISCUSSION THREAD

**Discussion by dcantos1** [Azure/ipam#285] [Ideas]
Hello, good afternoon, we are testing IPAM, it is super good but we also manage AWS IPAM and we could have points for improvement.

1. When the space and the block are created to be able to categorize the subnets, it would be good if it could be added so that they were overlapping because that way I can have everything in one space and see that I can debug without the need for it to remain uncategorized.

2. I do not have the visibility of the public IPs that I have, this is an important point, it would be great to have a panel to know how many public IPs I have displayed, which public IPs I have associated with a service and which not, to be able to debug.

3. The export to a CSV part is important.

4. In the documentation I don't know if there is a way to update the IPAM version without having to deploy one again and affect the configurations that are already managed.

**Comment by DCMattyG** [comment]
Hi @dcantos1, this is wonderful feedback. Please allow me to respond point-by-point...

1. The current design is that a **Space** can only contain non-overlapping IP space. I do agree there could be better ways remediate current brownfield environments where overlap exists, perhaps we can discuss a feature/tool/doc that might make this easier for you and other customers.
2. We current **do** expose Public IP's within the UI. See the screenshot below as an example, not sure if that helps your specific use case. I agree we could probably make this a proper column within the Endpoints section which would make it easier to view/search.
3. This is coming shortly, but there are ways around this using some light PowerShell until this feature is complete.
4. This depends on how you deploy the product. If you're using the standard App Service deployment, you only need to restart your App Service in order to complete the upgrade. No other actions are required.

Example of Viewing Public IPs:

![image](https://github.com/Azure/ipam/assets/12955732/074edfda-5588-482e-ad5d-f2e4c8c6462a)

![image](https://github.com/Azure/ipam/assets/12955732/0bcd747b-624b-4651-bb6c-1c25addbf5e5)

I hope the above helps a bit, always happy to chat live if you would prefer that as well. Have a great one!

**Comment by dcantos1** [comment]
Thank you very much friend for your response.

1. It would be good if it had an option that says YES or NO, do you want to add if it is overlapping, so it allows me to categorize it into one and see better.

2. In what you say, since I know which has a public IP and which does not, I can have other services that have public IP and suddenly they are associated or not and that would be good to have that visibility, that option is super good and is already provided by AWS .

3. Thank you, I hope to see you soon.

4. I understand that I just restart and it updates itself or I have to carry out some additional process.

I attach a screen where there is a query through the graph where I obtain the public addresses.

![image](https://github.com/Azure/ipam/assets/80493416/96f29347-cdc5-4415-a15c-caca8a2bdcd0)

**Reply by DCMattyG** [reply]
Thanks for that @dcantos1, here are my notes:

1. While this would be nice, this would require a significant architectural changes as to how the product currently works. I'll put some though into perhaps a planning tool within the product that can help make the migration process easier.
2. I agree, I'll put some thought into a way to better surface the Public IP utilization. If you have some screenshots as to how the AWS tool handles this, or a link to documentation I can reference, that would be helpful.
3. N/A
4. Yes, confirming that you **do not** need to re-deploy the solution, simply restart the App Service and the updated container will be downloaded & started.

Really appreciate all of the feedback and ideas!

**Comment by dcantos1** [comment]
Thanks for that @DCMattyG , the AWS docs IPAM public IP.

https://docs.aws.amazon.com/vpc/latest/ipam/view-public-ip-insights.html

**Reply by DCMattyG** [reply]
Thanks @dcantos1, so are you looking for something a long the lines of this visualization?

![image](https://github.com/Azure/ipam/assets/12955732/97cedc95-fcc9-4f15-874d-36d8d17e2b49)

And then perhaps details for each Public IP such as this?

![image](https://github.com/Azure/ipam/assets/12955732/049b9071-98ba-4780-ae13-5b7c4fbaef5e)

**Comment by dcantos1** [comment]
Hello @DCMattyG  is correct.
